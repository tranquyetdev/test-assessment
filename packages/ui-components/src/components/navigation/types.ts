export type NavigationProps = {
  menuItems: MenuItem[];
  currentPathname: string;
};

export type MenuItem = {
  title: string;
  url: string;
};
