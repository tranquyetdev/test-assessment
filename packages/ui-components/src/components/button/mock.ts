import { ButtonProps } from './button';

export const mockButtonProps: Record<ButtonProps['type'], ButtonProps> = {
  button: {
    type: 'button',
    children: 'button',
    onClick: () => console.log('hello world'),
  },
  link: {
    type: 'link',
    children: 'link',
    href: 'https://github.com/',
    title: 'Github',
  },
};
