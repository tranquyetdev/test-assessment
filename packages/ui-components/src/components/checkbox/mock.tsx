import { CheckBoxProps, defaultBgcolor } from '.';

export const mockCheckBoxProps: CheckBoxProps = {
  label: 'Developer',
  bgColor: defaultBgcolor,
  id: 'checkbox',
  checked: false,
  disabled: false,
};
