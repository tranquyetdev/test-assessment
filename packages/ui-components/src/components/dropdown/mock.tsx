export const mockDropdownProps = {
  content: (
    <ul className="menu">
      <li className="menu-item">
        <button>Menu 1</button>
      </li>
      <li className="menu-item">
        <button>Menu 2</button>
      </li>
    </ul>
  ),
  button: <div className="border">Show content</div>,
};
