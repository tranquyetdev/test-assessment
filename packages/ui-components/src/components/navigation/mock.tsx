import { NavigationProps } from './types';

export const mockNavigationProps: NavigationProps = {
  menuItems: [
    {
      title: 'Test Management',
      url: '/test-management',
    },
    {
      title: 'Candidate',
      url: '/candidate',
    },
    {
      title: 'Candidate',
      url: '/candidate',
    },
  ],
  currentPathname: '/test-management',
};
