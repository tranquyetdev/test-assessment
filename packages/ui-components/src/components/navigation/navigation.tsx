import * as React from 'react';
import { NavigationProps } from './types';
import Link from 'next/link';

export const Navigation = React.forwardRef((props: NavigationProps, ref) => {
  const { menuItems, currentPathname } = props;
  const styles = useStyles();

  return (
    <div className={`${styles.root}`}>
      {menuItems?.map((menuItem) => {
        return (
          <Link
            key={menuItem.url}
            className={`${styles.link(currentPathname === menuItem.url)}`}
            href={menuItem.url}
            data-title={menuItem.title}
          >
            {menuItem.title}
          </Link>
        );
      })}
    </div>
  );
});

const useStyles = () => {
  return {
    root: 'flex bg-neutral-text-primary text-13',
    link: (routeActive: boolean) =>
      `mx-[32px] py-[16px] border-b-2 text-center 
      ${
        routeActive
          ? 'border-primary-base text-neutral-white font-bold'
          : 'border-none text-neutral-disable hover:text-neutral-white hover:font-bold'
      } 
      hover:border-b-2 hover:border-primary-1 
      after:content-[attr(data-title)] after:block after:h-[0px] 
      after:invisible after:font-bold after:overflow-hidden`,
  };
};
