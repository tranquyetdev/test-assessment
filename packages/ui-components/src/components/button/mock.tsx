import { RenderIcon } from '../icons';
import { ButtonProps } from './button';

export const mockButtonProps: Record<ButtonProps['type'], ButtonProps> = {
  button: {
    type: 'button',
    children: 'button',
    onClick: () => alert('hello world'),
    LeftIcon: <RenderIcon name="caution" className="text-error" />,
    RightIcon: (
      <RenderIcon name="caution" className="text-error" />
    ),
  },
  link: {
    type: 'link',
    children: 'link',
    href: 'https://github.com/',
    title: 'Github',
    LeftIcon: <RenderIcon name="caution" className="text-error" />,
    RightIcon: (
      <RenderIcon name="caution" className="text-error" />
    ),
  },
};
