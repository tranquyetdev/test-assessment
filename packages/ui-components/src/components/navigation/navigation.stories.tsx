import type { Meta } from '@storybook/react';
import { Navigation as NavigationComponent } from './navigation';
import { mockNavigationProps } from './mock';

const Story: Meta<typeof NavigationComponent> = {
  title: 'Components / Navigation',
  component: NavigationComponent,
};
export default Story;

export const Navigation = () => (
  <NavigationComponent {...mockNavigationProps}></NavigationComponent>
);
