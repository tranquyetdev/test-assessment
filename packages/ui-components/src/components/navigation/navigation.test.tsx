import React from 'react';
import { render } from '@testing-library/react';
import { Navigation } from './navigation';
import { mockNavigationProps } from './mock';

describe('Navigation', () => {
  it(`should render Navigation successfully`, () => {
    const { baseElement } = render(<Navigation {...mockNavigationProps} />);
    expect(baseElement).toMatchSnapshot();
  });
});
