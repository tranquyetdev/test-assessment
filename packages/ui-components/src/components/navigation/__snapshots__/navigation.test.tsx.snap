// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render Navigation successfully 1`] = `
<body>
  <div>
    <div
      class="flex bg-neutral-text-primary text-13"
    >
      <a
        class="mx-[32px] py-[16px] border-b-2 text-center 
      border-primary-base text-neutral-white font-bold 
      hover:border-b-2 hover:border-primary-1 
      after:content-[attr(data-title)] after:block after:h-[0px] 
      after:invisible after:font-bold after:overflow-hidden"
        data-title="Test Management"
        href="/test-management"
      >
        Test Management
      </a>
      <a
        class="mx-[32px] py-[16px] border-b-2 text-center 
      border-none text-neutral-disable hover:text-neutral-white hover:font-bold 
      hover:border-b-2 hover:border-primary-1 
      after:content-[attr(data-title)] after:block after:h-[0px] 
      after:invisible after:font-bold after:overflow-hidden"
        data-title="Candidate"
        href="/candidate"
      >
        Candidate
      </a>
      <a
        class="mx-[32px] py-[16px] border-b-2 text-center 
      border-none text-neutral-disable hover:text-neutral-white hover:font-bold 
      hover:border-b-2 hover:border-primary-1 
      after:content-[attr(data-title)] after:block after:h-[0px] 
      after:invisible after:font-bold after:overflow-hidden"
        data-title="Stats"
        href="/stats"
      >
        Stats
      </a>
    </div>
  </div>
</body>
`;
